### PR TITLE
allow MetricCollection with args

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?

Allow passing metrics as arguments...
```python
# Example (input as arguments):
metrics = MetricCollection(
    Accuracy(),
    Precision(num_classes=3, average='macro'),
    Recall(num_classes=3, average='macro'),
)
```
reaction to https://github.com/PyTorchLightning/metrics/pull/175#issuecomment-821945333

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@PyTorchLightning/core-metrics and @justusschock @SkafteNicki what do you think?

File: torchmetrics/collections.py [hunk 1]
```diff
@@ -72,26 +79,37 @@ class MetricCollection(nn.ModuleDict):
...
+        elif isinstance(metrics, Sequence):
+            # prepare for optional additions
+            metrics = list(metrics)
+        if metric:
+            metrics += [m for m in metric if isinstance(m, Metric)]
+            remain = [m for m in metric if not isinstance(m, Metric)]
+            if remain:
+                rank_zero_warn(
+                    f"You have passes extra arguments {remain} which are not `Metric` so they will be ignored."
+                )
```
Comment (justusschock) [inline torchmetrics/collections.py:99]:
```suggestion
        remain = []
        for m in additional_metrics:
            if isinstance(m, Metric):
                metrics.append(m)
            else:
                remain.append(m)
                
        if remain:
            rank_zero_warn(
                f"You have passes extra arguments {remain} which are not `Metric` so they will be ignored."
            )
```

File: torchmetrics/collections.py [hunk 2]
```diff
@@ -72,26 +79,37 @@ class MetricCollection(nn.ModuleDict):
 
     def __init__(
         self,
-        metrics: Union[List[Metric], Tuple[Metric], Dict[str, Metric]],
+        metrics: Union[Metric, Sequence[Metric], Dict[str, Metric]],
+        *metric: Metric,
```
Comment (justusschock) [inline torchmetrics/collections.py:83]:
```suggestion
        *additional_metrics: Metric,
```

**Review by justusschock — APPROVED** [review]
I like it. just some comments for the naming and to avoid iterating over the metrics twice :)

File: torchmetrics/collections.py (hunk 2 above)
Comment (SkafteNicki) [inline torchmetrics/collections.py:83]: docstring missing for new arg

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**